packages/router/router.ts

```diff
@@ -2932,9 +2950,11 @@ function processRouteLoaderData(
   });
 
   // If we didn't consume the pending action error (i.e., all loaders
-  // resolved), then consume it here
+  // resolved), then consume it here.  Also clear out any loaderData for the
+  // throwing route
   if (pendingError) {
     errors = pendingError;
+    loaderData[Object.keys(pendingError)[0]] = undefined;
```

Comment:
If a route threw in an action, we wouldn't have run it's loaders so we don't hit it in the `results` loop above, so also ensure we clear out it's prior `loaderData`